homeassistant/components/scrape/__init__.py

```diff
@@ -60,7 +62,11 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
 
     for resource_config in scrape_config:
         if not (sensors := resource_config.get(SENSOR_DOMAIN)):
-            raise PlatformNotReady("No sensors configured")
+            _LOGGER.warning(
```

Comment:
Why isn't this config item required if we want to warn that it's missing?

Reply:
That was my initial thinking.
But if one resource config has sensors missing (for whatever reason) should it stop also the other ones from being created.

My conclusion was to print a warning and continue. What do you think?

Reply:
I think that all of the config should be correct for the integration. We don't do partial validation for other integrations that have config YAML still.

Reply:
I agree with @MartinHjelmare - I think we should be able to add the validation inside the schema to make it compulsory, and we can adjust later if it causes issues.

Reply:
Any suggestion on what that could look like. All my tries still allows the sensor to be an empty list.

Reply:
Maybe allowing an empty list isn't bad after all. `rest` allows it
It would mean that the coordinator gets created, but never refreshes afterward initialisation.

That would also allow people to debug their coordinator without the need to created initial sensors.

Reply:
Also with your last change you get the raw data from debug without having the sensors I guess it makes even more sense.
How do we proceed?

Reply:
This is what I have in mind (aligned with REST):
```python
async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
    """Set up Scrape from yaml config."""
    scrape_config: list[ConfigType] | None
    if not (scrape_config := config.get(DOMAIN)):
        return True

    refresh_coroutines: list[Coroutine[Any, Any, None]] = []
    load_coroutines: list[Coroutine[Any, Any, None]] = []
    for resource_config in scrape_config:
        rest = create_rest_data_from_config(hass, resource_config)
        coordinator = ScrapeCoordinator(
            hass,
            rest,
            timedelta(
                seconds=resource_config.get(CONF_SCAN_INTERVAL, DEFAULT_SCAN_INTERVAL)
            ),
        )

        refresh_coroutines.append(coordinator.async_refresh())

        sensors: list[ConfigType] | None = resource_config.get(SENSOR_DOMAIN)
        for sensor_config in sensors or []:
            load_coroutines.append(
                discovery.async_load_platform(
                    hass,
                    Platform.SENSOR,
                    DOMAIN,
                    {"coordinator": coordinator, "config": sensor_config},
                    config,
                )
            )

    if refresh_coroutines:
        await asyncio.gather(*refresh_coroutines)

    if load_coroutines:
        await asyncio.gather(*load_coroutines)

    return True
```